doc/nrf/app_memory.rst

```diff
@@ -84,6 +84,48 @@ Besides applying `General recommendations`_, you can also complete the following
 * Reduce the verbosity of assert messages by setting :kconfig:option:`CONFIG_ASSERT_VERBOSE` to ``n``.
 * Check `Thread`_ memory footprint optimization actions, as the Matter application layer uses OpenThread.
 
+.. _app_memory_nfc:
+
+NFC
+***
+
+Tag
+---
+
+To optimize your application that supports the NFC tag functionality follow the `General recommendations`_.
+The NFC :ref:`type_2_tag` and :ref:`type_4_tag` libraries do not provide configuration options that have an effect on memory usage in an application.
+However, there are a few Kconfig configuration options you can use to optimize memory usage related to NFC.
+
+* :kconfig:option:`CONFIG_NFC_PLATFORM_LOG_LEVEL_CHOICE` to reduce logging level in the NFC integration module.
+
+For an application that uses the :ref:`type_4_tag` library, you can set the following options:
+
+* :kconfig:option:`CONFIG_NDEF_FILE_SIZE` for the maximum NDEF file size, if the read-write mode is supported,
+* :kconfig:option:`CONFIG_NFC_TNEP_RX_MAX_RECORD_CNT` and :kconfig:option:`CONFIG_NFC_TNEP_RX_MAX_RECORD_SIZE` for the maximum buffer size of NDEF message exchange, if the TNEP protocol is supported.
+
+.. _tnep_ch_optimization:
+
+If the application uses the NFC TNEP protocol for the Bluetooth LE out-of-band pairing procedure (:ref:`nfc_tnep_ch_readme`), you can set the following options:
+
+* :kconfig:option:`CONFIG_NFC_TNEP_CH_MAX_RECORD_COUNT`
+* :kconfig:option:`CONFIG_NFC_TNEP_CH_MAX_LOCAL_RECORD_COUNT`
+* :kconfig:option:`CONFIG_NFC_TNEP_CH_PARSER_BUFFER_SIZE`
+
+Poller
+------
+
+To optimize an application that supports the NFC poller functionality using the :ref:`st25r3911b_nfc_readme` library, you can set the following options:
+
+* :kconfig:option:`CONFIG_ST25R3911B_LIB_LOG_LEVEL_CHOICE` and similar options to reduce the logging level of the NFC components used in the application.
+* :kconfig:option:`CONFIG_NFC_T4T_HL_PROCEDURE_CC_BUFFER_SIZE` and :kconfig:option:`CONFIG_NFC_T4T_HL_PROCEDURE_APDU_BUF_SIZE` to adjust the buffer sizes in the NFC T4T protocol implementation of the poller side.
+
+If the application supports the TNEP protocol, you can set the following options:
+
+* :kconfig:option:`CONFIG_NFC_TNEP_POLLER_RX_MAX_RECORD_CNT` for the maximum number of NDEF records in the received message.
+* :kconfig:option:`CONFIG_NFC_TNEP_CH_POLLER_RX_BUF_SIZE` for the Connection Handover receive buffer size of the poller mode, if the application uses the :ref:`nfc_tnep_ch_readme` library.
+
+:ref:`tnep_ch_optimization`
```

Comment:
I decided to divide the text into 2 section: tag and poller, so it is clear when to apply which optimizations. The broken link comes from  the fact that the last section: _If the application uses the NFC TNEP protocol for the Bluetooth LE out-of-band pairing procedure..._ applies to both tag and poller, so not sure how to solve it. I reversed it back to the original form. Do you have any suggestion how it can be solved so it's clear the last section is for both NFC roles?